### PR TITLE
Fixes #147 creates UI for Advanced section of Control UI

### LINKED ISSUE #147: Control UI Implementation - Advanced
The parent issue is here #41 . The parent issue has some important discussions regarding the UI design. This thread contains the functionality of using PSLab as an Arbitrary Waveform Generator, PWM configuration & for digital outputs. 

-  [ ] Layout design
-  [ ] Code for adding functionalities

The mockup for the same is here.

The actual UI is here 

The design in the UI is not final and might change in the PR.

### PR DESCRIPTION
**Pull request by asitava1998**
Fixes issue #147 

Changes: 
* Created a custom component using EditText, increment & decrement buttons to improve code quality & efficiency.
* Enabled functionality to the buttons of Advanced section of Control UI. Only scienceLab method calls remain to be added.
* Changed input type of edittext in `fragment_control_advanced.xml` to decimal.
* Enabled scrollview in read section of Control UI since in landscape mode, the entire layout was not covered.
* Added the standard dimensions in dimens.xml

Screenshots for the change: 
Change in UI due to addition of scrollview in Control Read

### CODE REVIEW

**Comment by viveksb007** [comment]
@asitava1998 I think it's not efficient and decreases code quality using separate buttons for increment and decrement and placing everything discretely. Have a look at how to create custom views and custom components. https://developer.android.com/guide/topics/ui/custom-components.html
After that try to incorporate two buttons and single edit text in common view and that view would have functionality to increment and decrement their corresponding edit text.
This would make code concise and clear.
Thanks.

**Comment by asitava1998** [comment]
@viveksb007  I also feel that would be better. Let me then create a custom widget and then use it all the places.

File: app/src/main/res/layout/fragment_control_advanced.xml [hunk 1]
```diff
@@ -619,7 +631,8 @@
                     android:text="@string/sqr2"
                     android:textSize="18sp"
                     android:textStyle="bold"
-                    android:layout_marginStart="12dp"/>
+                    android:layout_marginStart="12dp"
```
Comment (CloudyPadmal) [inline app/src/main/res/layout/fragment_control_advanced.xml]: These values can be added to `dimens.xml` and refer them in here! :)

Comment (asitava1998) [inline app/src/main/res/layout/fragment_control_advanced.xml]: Yes. I will do it along with other changes. Actually, I will be replacing lot of the repetitive stuff by creating a custom component.

Comment (CloudyPadmal) [inline app/src/main/res/layout/fragment_control_advanced.xml]: That's great! :+1: :blush:

**Comment by viveksb007** [comment]
@asitava1998 Could you update screenshots with custom view components?

**Comment by asitava1998** [comment]
@viveksb007  For creating the custom view, I used the edittext and increment & decrement buttons beside it. So, the UI is essentially the same with the custom view.

**Comment by viveksb007** [comment]
Text inside spinner isn't readable. Edit text is taking way more space compared to adjacent buttons. Take care of such things and try that UI looks uniform and UX should be smooth. Use proper padding to separate one element to another.  I find buttons quite small and difficult to prevent clicking **+** instead of **-**

**Comment by asitava1998** [comment]
Here is the link to the [apk](https://drive.google.com/file/d/0B3F6wGD_1dcZdUhubzdZRjM2LWM/view?usp=sharing) .

@viveksb007 @akarshan96 @CloudyPadmal @jithinbp Please check the control part and suggest overall changes in UI.

**Comment by viveksb007** [comment]
@asitava1998 the app crashes when button is clicked without giving a value to editText.
Click those purple **+** , **-** buttons without writing any value in EditText it crashes. 
And buttons size is not up to the mark yet. Try to make its height equal to EditText and corresponding width to make it a square.

**Comment by asitava1998** [comment]
@viveksb007 Here is the link of the [apk](https://drive.google.com/open?id=0B3F6wGD_1dcZS3RNeWhtZGgzS2M) . The app crashing problem has been fixed and I have uploaded screenshots in the main description of the layouts.

**Comment by viveksb007** [comment]
gravity in advanced isn't center and editText lower edge looks like its been covered under some other view element. Have a look please and give apk for testing after making sure you have covered every corner.

screenshot_2017-06-22-23-41-12-808_org fossasia pslab

**Comment by CloudyPadmal** [comment]
Hello @asitava1998 ! I think it would look better if we have a similar text size at every section.

ui

**Comment by asitava1998** [comment]
@CloudyPadmal  I will fix this and I notice that the Advanced icon from the bottom bar has disappeared for some reason, I have to check that too. What is your API level?

**Comment by asitava1998** [comment]
@CloudyPadmal @akarshan96 @jithinbp @viveksb007  Here is the updated [apk](https://drive.google.com/file/d/0B3F6wGD_1dcZMXlndDBDeEp6eTQ/view?usp=sharing) . I have made the changes pointed out by Vivek & Padmal in the last comment. 
I think, its better to merge this PR and create new issues if anyone still finds a bug. There might still be issues with very low resolution screens else for medium & high resolution, the display is quite fine.

**Comment by CloudyPadmal** [comment]
@asitava1998 the fix is great! All the issues were addressed it seems and the app works fine in my device. I have little question. Do we need a high resolution to `PV` values like the values in the screen below?

screenshot_2017-06-24-08-00-45

Or should I open a new issue as this PR has been here for a few days? :)

**Comment by asitava1998** [comment]
@CloudyPadmal Better if you open a new issue and fix it. The PR has been lying for long now.

**Comment by CloudyPadmal** [comment]
Sure thing @asitava1998 :+1:

**Pull request merged**